### ISSUE TITLE
Accessibility: macOS On-screen keyboard word completion issues

### ISSUE THREAD

**Issue by ZiedYousfi** [zed-industries/zed#38052]
### Summary

On macOS, On-screen keyboard completion bug 

### Description
Steps to trigger the problem:
1. Start typing in agent writing box with the On-screen keyboard
2. Choose a word in the On-screen keyboard completion

**Expected Behavior**: Type the full word
**Actual Behavior**: Last character missing

### Zed Version and System Specs

Zed: v0.203.4 (Zed) 
OS: macOS 15.6.1
Memory: 36 GiB
Architecture: aarch64

**Comment by notpeter** [comment]
I can reproduce. Enable this keyboard with:
System Preferences -> Accessibility -> Keyboard (in Motor Section) -> Accessibility Keyboard (enable).

When I type `Wh` it suggests `What` but clicking the word inserts `Wha `.
Subsequent completions are also broken. So instead of `What have`, clicking the `have` suggestion results in `Whahav ` (truncation and word is inserted before the space).

Thanks for reporting.

**Comment by inayayousfi** [comment]
It seems to have this behavior in every type of buffers.

**Comment by warrenjokinen** [comment]
> It seems to have this behavior in every type of buffers.

I don't see that behavior but I have other problems.
I am new to Accessibility Keyboard
So maybe my preferences are new maybe you have conflict with some existing software
I used voice recognition on my Mac desktop to enter this comment

**Comment by criticic** [comment]
@esthertrapadoux can reproduce this issue on Zed 0.225.10 as well, I am planning to work on it, as well as test the fix on linux and windows as well to ensure no regressions, is this open for contribution?

**Comment by esthertrapadoux** [comment]
Nice find @criticic ! And welcome to the Guild. Adding this to the board and assigning it to you!
